src/series/groupedSeries.js

```diff
@@ -1,31 +1,31 @@
-import barSeries from './bar';
 import d3 from 'd3';
 import dataJoinUtil from '../util/dataJoin';
 import {noop} from '../util/fn';
 import fractionalBarWidth from '../util/fractionalBarWidth';
 
-export default function() {
+export default function(series) {
 
-    var bar = barSeries(),
-        barWidth = fractionalBarWidth(0.75),
+    var barWidth = fractionalBarWidth(0.75),
         decorate = noop,
         xScale = d3.scale.linear(),
         offsetScale = d3.scale.linear(),
-        values = function(d) { return d.values; };
+        values = function(d) { return d.values; },
+        x = function(d, i) { return xScale(value(d, i)); },
```

Comment:
As the `x` function is only used once, I do wonder about the value of having it pulled up into this outer scope? Typically the variables found here are exposed as properties.